lib/modules/txt.mjs

```diff
@@ -343,3 +343,26 @@ export async function validateCopyrightNoticeSectionIsNumbered (doc, { mode = MO
 
   return result
 }
+
+/**
+ * Validate document name
+ *
+ * @param {Object} doc Document to validate
+ * @param {Object} [opts] Additional options
+ * @param {number} [opts.mode=0] Validation mode to use
+ * @returns {Array} List of errors/warnings/comments or empty if fully valid
+ */
+export async function validateDocumentName (doc, { mode = MODES.NORMAL } = {}) {
+  const result = []
+
+  if (!doc.data.slug) {
+    result.push(new ValidationError(
+      'DOCUMENT_NAME_MISSING',
+      'Document name missing or invalid.',
```

Comment:
idnits2 said:
>		errg("Expected the document's filename to be given on the first page, but didn't find any");
idnits3 should say
"Document's name not found on the first page".